gateway/openapi.yaml

```diff
@@ -0,0 +1,327 @@
+swagger: '2.0'
+info:
+  version: 1.0.0
+  title: Supplementary Data Service (SDS)
+  description: API gateway for SDS
+schemes:
+  - https
+securityDefinitions:
+  UserSecurity:
+    type: basic
+security:
+  - UserSecurity: []
+paths:
+  /v1/unit_data:
+    post:
+      description: retrieve data for a unit
+      parameters:
+        - name: dataset_id
+          in: query
+          required: true
+          type: string
+        - name: unit_id
+          in: query
+          required: true
+          type: string
+      responses:
+        200:
+          description: Succesfully retrieved data
+          schema:
+            type: object
+            properties:
+              status:
+                type: string
+              data:
+                type: object
+                properties:
+                  local_unit:
+                    type: object
+                      properties:
+                        luname:
+                          type: string
+                        luref:
+                          type: string
+                  ruref:
+                    type: string
+                  runame:
+                    type: string
+        400:
+          description: Bad request
+          schema:
+            $ref: "#/definitions/error_response"
+        404:
+          description: Item not found
+          schema:
+            $ref: "#/definitions/error_response"
+        500:
+          description: Internal error
+          schema:
+            $ref: "#/definitions/error_response"
+  /v1/schema:
+    post:
+      description:
+      parameters:
+        - name: schema
+          in: body
+          required: true
+          type: object
+          schema: "#/definitions/sds_schema"
+      responses:
+        200:
+          description: Successfuly submitted schema
+          schema:
+            type: object
+            schema:
+              $ref: "#/definitions/sds_schema_metadata"
+        400:
+          description: Schema validation failed
+          schema:
+            $ref: "#/definitions/error_response"
+        500:
+          description: Internal error
+          schema:
+            $ref: "#/definitions/error_response"
+    get:
+      description: return schema
+      parameters:
+        - name: survey_id
+          in: query
+          type: string
+          required: true
+        - name: version
+          in: query
+          type: string
+          required: true
+      responses:
+        200:
+          description: Succesfully retrieved schema
+          schema:
+            type: object
+            schema: "#/definitions/sds_schema"
+        404:
```

Comment:
This endpoint will also trigger a 400 response, for example when providing an invalid version

Reply:
Thanks will just add that in now

Reply:
The status code should be 400